### ISSUE TITLE
No such method error in 5.3 for an old model

### ISSUE THREAD

**Issue by reyman** [NetLogo/NetLogo#910]
When i launch my model realized for an old version of netlogo (2003) in the latest version 5.3, i have this error :

```
java.lang.NoSuchMethodError: org.nlogo.api.AgentSet.kind()Lorg/nlogo/api/AgentKind;
 at org.nlogo.extensions.network.Helpers$class.requireTurtleset(NetworkExtension.scala:30)
 at org.nlogo.extensions.network.MeanLinkPathLength$.requireTurtleset(NetworkExtension.scala:123)
 at org.nlogo.extensions.network.MeanLinkPathLength$.report(NetworkExtension.scala:131)
 at org.nlogo.prim._externreport.report(_externreport.java:50)
 at org.nlogo.prim._asm_procedurecomputeaverageshortestpathlength_setobservervariable_0.perform()
 at org.nlogo.nvm.Context.stepConcurrent(Context.java:91)
 at org.nlogo.nvm.ConcurrentJob.step(ConcurrentJob.java:82)
 at org.nlogo.job.JobThread.org$nlogo$job$JobThread$$runPrimaryJobs(JobThread.scala:143)
 at org.nlogo.job.JobThread$$anonfun$run$1.apply$mcV$sp(JobThread.scala:78)
 at org.nlogo.job.JobThread$$anonfun$run$1.apply(JobThread.scala:76)
 at org.nlogo.job.JobThread$$anonfun$run$1.apply(JobThread.scala:76)
 at scala.util.control.Exception$Catch.apply(Exception.scala:88)
 at org.nlogo.util.Exceptions$.handling(Exceptions.scala:41)
 at org.nlogo.job.JobThread.run(JobThread.scala:75)

NetLogo 5.3
main: org.nlogo.app.AppFrame
thread: AWT-EventQueue-0
Java HotSpot(TM) 64-Bit Server VM 1.8.0_65 (Oracle Corporation; 1.8.0_65-b17)
operating system: Windows 10 10.0 (amd64 processor)
Scala version 2.9.2
JOGL: (3D View not initialized)
OpenGL Graphics: (3D View not initialized)
model: EpiSim_Modele

12:59:09.050 TickStateChangeEvent (org.nlogo.app.App$$anon$1 (org.nlogo.window.GUIWorkspace)) JobThread
12:59:08.880 InterfaceGlobalEvent (org.nlogo.app.InterfacePanel$2 (org.nlogo.window.SliderWidget)) AWT-EventQueue-0
12:59:08.879 InterfaceGlobalEvent (org.nlogo.app.InterfacePanel$2 (org.nlogo.window.SliderWidget)) AWT-EventQueue-0
12:59:08.879 InterfaceGlobalEvent (org.nlogo.app.InterfacePanel$2 (org.nlogo.window.SliderWidget)) AWT-EventQueue-0
12:59:08.879 InterfaceGlobalEvent (org.nlogo.app.InterfacePanel$2 (org.nlogo.window.SliderWidget)) AWT-EventQueue-0
12:59:08.879 InterfaceGlobalEvent (org.nlogo.window.ChooserWidget) AWT-EventQueue-0
12:59:08.879 InterfaceGlobalEvent (org.nlogo.app.InterfacePanel$2 (org.nlogo.window.SliderWidget)) AWT-EventQueue-0
12:59:08.879 InterfaceGlobalEvent (org.nlogo.app.InterfacePanel$2 (org.nlogo.window.SliderWidget)) AWT-EventQueue-0
12:59:08.879 InterfaceGlobalEvent (org.nlogo.app.InterfacePanel$2 (org.nlogo.window.SliderWidget)) AWT-EventQueue-0
12:59:08.879 InterfaceGlobalEvent (org.nlogo.window.ChooserWidget) AWT-EventQueue-0

```

**Comment by mrerrormessage** [comment]
Thanks for reporting this bug. It appears that the network extension that was released with 5.3 was broken. It appears that this bug was also present in 5.2.1, but you should be able to run the model in 5.2. The good news is that 5.2 and 5.3 are almost identical. The differences are primarily in changes to the models library. You can also copy the network extension from the 5.2 extensions folder to the 5.3 extensions folder and that should resolve the issue. We will look at getting this cleaned up in the next release, although it's unclear when that will happen.

**Comment by SethTisue** [comment]
@reyman the `network` extension is legacy at this point; it may not be included in future versions of NetLogo. I would suggest migrating the model to the newer, actively maintained `nw` extension

**Comment by reyman** [comment]
@SethTisue @mrerrormessage Thanks for your help :)

**Comment by mrerrormessage** [comment]
This will be included in the next 5.3 release, date TBD.
